hack/build-and-push.sh

```diff
@@ -431,6 +436,45 @@ attach_migration_file() {
     return 0
 }
 
+# Find previous bundle that has a migration and output its digest.
+find_previous_migration_bundle_digest() {
+    local -r task_name=${1:?Missing task name}
+    local -r task_version=${2:?Missing task version}
+
+    local limit=1
+    if [ -n "$TEST_REPO_NAME" ]; then
+        limit=10
+    fi
+
+    local bundle_ref
+    bundle_ref=$(generate_tagged_task_bundle "$task_name" "$task_version")
+    local ns_repo=${bundle_ref%:*}  # remove tag
+    ns_repo=${ns_repo#*/}  # remove registry
+    # Retain the tag used for matching tag names
+    local -r name_pattern="${bundle_ref##*:}-"
```

Comment:
Do we need it for non testing use-case?

Reply:
The tag generated by `generate_tagged_task_bundle` is affected by the `TEST_REPO_NAME` variable. It covers both.